src/views/common/Animation/AnimationUtilities/useTimeSyncing.ts

```diff
@@ -0,0 +1,122 @@
+import React, { useCallback, useMemo, useRef } from "react"
```

Comment:
This is currently used with the focus time from the recording, but it could be used for any frame that contains time data and external time source/sink. The question is which way to run the updates.